dotnet/src/SemanticKernel.Abstractions/AI/AIFunctionResultExtensions.cs

```diff
@@ -21,15 +21,30 @@ public static class AIFunctionResultExtensions
     /// <param name="result">Instance of <see cref="FunctionResult"/> class.</param>
     public static IReadOnlyCollection<ModelResult>? GetModelResults(this FunctionResult result)
     {
-        if (result.Metadata.TryGetValue(ModelResultsMetadataKey, out object modelResultObject) &&
-            modelResultObject is IReadOnlyCollection<ModelResult> modelResults)
+        if (result.TryGetValue(ModelResultsMetadataKey, out IReadOnlyCollection<ModelResult>? modelResults))
         {
             return modelResults;
         }
 
         return null;
     }
 
+    /// <summary>
+    /// Get typed value from <see cref="FunctionResult"/> metadata.
+    /// </summary>
+    public static bool TryGetValue<T>(this FunctionResult result, string key, out T value)
```

Comment:
Since `FunctionResult` already contains method `GetValue<T>`, it looks like this method `TryGetValue<T>` works in similar way, but with boolean result and `out` parameter. While it's working differently, since it returns value not from `object Value`, but from `Metadata`. So, the naming may be confusing here. I would name it `TryGetMetadata` or `TryGetMetadataValue` to clearly define that we work with `Metadata` here.